Refresh task on create job; provide manual refresh

apennell opened the pull request:
To provide a more transparent experience when creating a new job:
  - Display error toast if unable to load translations or create job due to missing `context` or if error is thrown while creating task
  - Set loading text on "Create job" button while processing
  - Show success toast on successful job creation
  - After successful job creation, reset form and get refreshed task
  - Because, in my testing experience, the newly added locale isn't always available immediately upon job creation, provide a "Refresh status" button so the user can grab it fresh themselves, without reloading page.

![Screen Shot 2022-02-16 at 5 49 52 PM](https://user-images.githubusercontent.com/11039888/154391702-3c5f1cab-6215-49c6-aad7-befd5eca72a9.png)

apennell committed:
Refresh task on create job; provide manual refresh

To provide a more transparent experience when creating a new job:
  - Display error toast if unable to load translations or create job due
    to missing `context` or if error is thrown while creating task
  - Set loading text on "Create job" button while processing
  - Show success toast on successful job creation
  - After successful job creation, reset form and get refreshed task
  - Because, in my testing experience, the newly added locale isn't
  always available immediately upon job creation, provide a "Refresh
  status" button so the user can grab it fresh themselves, without
  reloading page.

sanches89 reviewed (approved):
Great work :+1:

sanches89 commented on src/components/NewTask.tsx:
```diff
@@ -78,23 +79,56 @@ export const NewTask = ({ locales }: Props) => {
+          description: errorMsg,
+          status: 'error',
+          closable: true,
+        })
```
Just a comment, if the `catch` function is called before the `setIsBusy(false)` the `isBusy` will get stuck as `true`...
In this case you have two options, maybe more :smile::
- call `setIsBusy(false)` at the `catch` function too;
- remove the  `setIsBusy(false)` at the `then` function and add in the final of the promise chain a [function called `finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) and add the  `setIsBusy(false)`

sanches89 commented on src/components/NewTask.tsx:
```diff
@@ -78,23 +79,56 @@ export const NewTask = ({ locales }: Props) => {
         )
-      })
+      )
       .then(() => {
+        toast.push({
+          title: 'Job successfully created',
+          status: 'success',
+          closable: true,
+        })
+
+        /** Reset form fields */
+        setSelectedLocales([])
+        setSelectedWorkflowUid('')
         setIsBusy(false)
+
+        /** Update task data in TranslationView */
+        refreshTask()
+      })
+      .catch(err => {
+        let errorMsg
+        if (err instanceof Error) {
+          errorMsg = err.message
+        } else {
+          errorMsg = err ? String(err) : null
+        }
+
+        toast.push({
+          title: `Error creating translation job`,
+          description: errorMsg,
+          status: 'error',
+          closable: true,
+        })
       })
```
Just a comment, if the `catch` function is called before the `setIsBusy(false)` the `isBusy` will get stuck as `true`...
In this case you have two options, maybe more :smile::
- call `setIsBusy(false)` at the `catch` function too;
- remove the  `setIsBusy(false)` at the `then` function and add in the final of the promise chain a [function called `finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) and add the  `setIsBusy(false)`

apennell replied:
Thanks for catching that and for recommending `finally` :)

apennell committed:
Reset button in finally

Reset button to not busy in `finally` so it doesn't get stuck disabled
if error is caught
